packages/cli/src/databases/migrations/mysqldb/1690000000001-MigrateIntegerKeysToString.ts

```diff
@@ -1,38 +1,55 @@
 import type { MigrationContext, ReversibleMigration } from '@db/types';
 
+const COLLATION_57 = 'utf8mb4_general_ci';
+const COLLATION_80 = 'utf8mb4_0900_ai_ci';
+
 export class MigrateIntegerKeysToString1690000000001 implements ReversibleMigration {
 	async up({ queryRunner, tablePrefix }: MigrationContext) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const dbVersionQuery = (await queryRunner.query('SELECT @@version')) as  // eslint-disable-next-line @typescript-eslint/naming-convention
+			| Array<{ '@@version': string }>
+			| undefined;
+		let collation = COLLATION_80;
+		if (dbVersionQuery?.length === 1) {
```

Comment:
[we have a similar check in a previous migration](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/databases/migrations/mysqldb/1623936588000-CertifyCorrectCollation.ts#L10). maybe we can unify the two, or at least use the same code in both the places.